service/src/main/java/org/eclipse/keti/acs/service/policy/evaluation/PolicyEvaluationServiceImpl.java

```diff
@@ -95,27 +93,11 @@ public PolicyEvaluationResult evalPolicy(final PolicyEvaluationRequestV1 request
                     + "Please review and resubmit the request.");
         }
 
-        List<PolicySet> allPolicySets = this.policyService.getAllPolicySets();
-
-        if (allPolicySets.isEmpty()) {
-            return new PolicyEvaluationResult(Effect.NOT_APPLICABLE);
-        }
-
-        LinkedHashSet<PolicySet> filteredPolicySets = filterPolicySetsByPriority(subjectIdentifier, uri, allPolicySets,
-                policySetsEvaluationOrder);
-
-        // At this point empty evaluation order means we have only one policy set.
-        // Fixing policy evaluation order so we could build a cache key.
-        PolicyEvaluationRequestCacheKey key;
-        if (policySetsEvaluationOrder.isEmpty()) {
-            key = new Builder().zoneId(zone.getName())
-                    .policySetIds(Stream.of(filteredPolicySets.iterator().next().getName())
-                            .collect(Collectors.toCollection(LinkedHashSet::new)))
-                    .request(request).build();
-        } else {
-            key = new Builder().zoneId(zone.getName()).request(request).build();
-        }
-
+        // When policySetEvaluationOrder is not specified in the request, cached decision is invalidated if any
```

Comment:
This comment is better placed in the Key constructor where the special value is assigned